daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -897,6 +897,33 @@ func (devices *DeviceSet) createBaseImage() error {
 	return nil
 }
 
+// Returns if thin pool device exists or not. If device exists, also makes
+// sure it is a thin pool device and not some other type of device.
+func (devices *DeviceSet) thinPoolExists(thinPoolDevice string) (bool, error) {
+	logrus.Debugf("devmapper: Checking for existence of the pool %s", thinPoolDevice)
+
+	info, err := devicemapper.GetInfo(thinPoolDevice)
+	if err != nil {
+		return false, fmt.Errorf("devmapper: GetInfo() on device %s failed: %v", thinPoolDevice, err)
+	}
+
+	// Device does not exist.
+	if info.Exists == 0 {
+		return false, nil
+	}
+
+	_, _, deviceType, _, err := devicemapper.GetStatus(thinPoolDevice)
+	if err != nil {
+		return true, fmt.Errorf("devmapper: GetStatus() on device %s failed: %v", thinPoolDevice, err)
```

Comment:
this signature slightly bother me, let's return false(default bool value) on errors, that value shouldn't be used in case of error anyway.